doc/introduction/remote.rst

```diff
@@ -228,104 +154,122 @@ array([[0, 0, 1, 0, 1, 0, 1, 0],
 array([0.4 , 0.1 , 0.15, 0.05, 0.3 , 0.3 , 0.45, 0.35])
 
+We can also use the Python collections module to convert the samples into
+counts:
 
-We can convert the samples into counts using the following function:
+>>> from collections import Counter
+>>> bitstrings = [tuple(i) for i in results.samples]
+>>> counts = {k:v for k, v in Counter(bitstrings).items()}
+>>> counts[(0, 0, 0, 0, 0, 0, 0)]
+2
 
-.. code-block:: python3
+.. note::
 
-     from collections import Counter
+    The :class:`~.operation` decorator allows you to create your own Strawberry Fields
+    operation. This can make it easier to ensure that the same unitary is always
+    applied to the signal and idler modes.
 
-     def count(samples):
-          bitstrings = [tuple(i) for i in samples]
-          return {k:v for k, v in Counter(bitstrings).items()}
+    .. code-block:: python3
 
->>> samples = np.array([[0, 2], [1, 0], [0, 1], [0, 0], [0, 0], [2, 0], [0, 1], [0, 1]])
->>> counts = count(samples)
->>> print(counts)
-{(0, 2): 1, (1, 0): 1, (0, 1): 3, (0, 0): 2, (2, 0): 1}
->>> counts[(0, 0)]
-2
+        from strawberryfields.utils import operation
 
-.. _compilation:
+        @operation(4)
+        def unitary(q):
+            ops.Interferometer(U) | q
+            ops.BSgate(0.543, 0.123) | (q[2], q[0])
+            ops.Rgate(0.453) | q[1]
+            ops.MZgate(0.65, -0.54) | (q[2], q[3])
 
-Program compilation
--------------------
+        prog = sf.Program(8)
 
-In addition to using the program template above, which directly matches the physical
-layout of the hardware device, you can apply any four-mode interferometer to the pairs of modes.
+        with prog.context as q:
+            ops.S2gate(1.0) | (q[0], q[4])
+            ops.S2gate(1.0) | (q[1], q[5])
+            ops.S2gate(1.0) | (q[3], q[7])
 
-Primitive gates supported by ``chip2`` include any combination of:
+            unitary() | q[:4]
+            unitary() | q[4:]
+ 
+            ops.MeasureFock() | q
 
-* `General beamsplitters <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.BSgate.html>`_ (:class:`~.ops.BSgate`),
+    Refer to the :class:`~.operation` documentation for more details.
 
-* `Mach-Zehnder interfomerters <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.MZgate.html>`_ (:class:`~.ops.MZgate`), or
+Managing submitted jobs
+-----------------------
 
-* `rotations/phase shifts <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.Rgate.html>`_ (:class:`~.ops.Rgate`).
+In the above example, :meth:`RemoteEngine.run` is a **blocking** method; Python will wait until
+the job result has been returned from the cloud before further lines
+will execute.
 
-Furthermore, several automatic decompositions are supported:
+**Non-blocking jobs** can also be submitted using the :meth:`eng.run_async() <~.RemoteEngine.run_async>` method. Unlike ``eng.run()``, this method returns a :class:`~.Job`
```

Comment:
Suggest rewording. It's not really the job that's blocking here, it's the call

Reply:
Rephrased to:
```rest
Jobs can also be submitted using the **non-blocking**
:meth:`eng.run_async() <.RemoteEngine.run_async>` method.
```